### PR TITLE
Transport layer for proxy handshake

### PR DESCRIPTION
**Pull request by SreeramGarlapati**
@gemmellr @tabish121 - truly appreciate - if you can review this.

cc: @timtay-microsoft @apraovjr @jongio @JamesBirdsall @binzywu @yvgopal @ShubhaVijayasarathy @tameraw @prmathur-microsoft @jasmineymlo @nemakam

[Tracking Issue](https://github.com/Azure/azure-event-hubs-java/issues/264).

### CODE REVIEW

File: src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java [hunk 1]
```diff
@@ -94,6 +94,7 @@ public void configure(
             WebSocketHandler webSocketHandler) {
         this.host = host;
         this.path = path;
+        this.query = query;
```
Comment (zolvarga) [inline src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java]: >query [](start = 13, length = 5)

Good catch

File: src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    private final ByteBuffer inputBuffer;
+    private final ByteBuffer outputBuffer;
+
+    private boolean tailClosed = false;
+    private boolean headClosed = false;
+    private boolean isProxyConfigured;
+    private String host = "";
+    private Map<String, String> headers = null;
+    private TransportImpl underlyingTransport;
+    private ProxyState proxyState = ProxyState.PN_PROXY_NOT_STARTED;
```
Comment (zolvarga) [inline src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java]: This is just a question:
What are we doing after the state machine ended up in FAILURE state? We would never go back to NOT_STARTED, is that right? What is the requirement for the user to start again?

Comment (SreeramGarlapati) [reply src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java]: That's right - @zolvarga. We wouldn't go back to NOT_STARTED. 
Intention was to signal - that this transport is closed.

Comment (SreeramGarlapati) [reply src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java]: The flow goes like this: 
 * once we set `TransportImpl` to `FAILED` state - proton - invokes `onTransportError()` - with the response from `Proxy` - and closes the transport. 
 * Then, developer can make a choice based on the response - whether to retry - by recreating the connection or simply throw to the layers above.

**Pull request merged**